frontend/packages/console-shared/src/utils/pod-ring-utils.ts

```diff
@@ -69,23 +68,28 @@ export const usePodScalingAccessStatus = (
   obj: K8sResourceKind,
   resourceKind: K8sKind,
   pods: ExtPodKind[],
+  apiCallfn: (
+    res: K8sResourceKind,
+    kind: K8sKind,
+    impersonate: string,
+  ) => Promise<SelfSubjectAccessReviewKind>,
```

Comment:
This is odd for client code to have to provide this isn't it?
Seems like you are just doing this change so that you can unit test the code.
Why not mock `checkPodEditAccess` instead?